Does not work on Linux due to "recursive watch"

**Describe the bug**
A clear and concise description of what the bug is. If there is an error in the inspector pane, please paste that string in.

Using npm 16, or any version above 13, gives (with ``yarn start``):


```
@penrose/core: $ yarn build:parsers
@penrose/browser-ui: $ node build.js -w
@penrose/browser-ui:  > src/App.tsx:15:7: error: Could not resolve "@penrose/core" (mark it as external to exclude it from the bundle)
@penrose/browser-ui:     15 │ } from "@penrose/core";
@penrose/browser-ui:        ╵        ~~~~~~~~~~~~~~~
@penrose/browser-ui: serving ./public at http://localhost:3000/ (livereload on :13000)
@penrose/browser-ui:  > src/inspector/views/Errors.tsx:1:26: error: Could not resolve "@penrose/core" (mark it as external to exclude it from the bundle)
@penrose/browser-ui:     1 │ import { showError } from "@penrose/core";
@penrose/browser-ui:       ╵                           ~~~~~~~~~~~~~~~
@penrose/browser-ui:  > src/inspector/views/GraphUtils.ts:6:7: error: Could not resolve "@penrose/core" (mark it as external to exclude it from the bundle)
@penrose/browser-ui:     6 │ } from "@penrose/core";
@penrose/browser-ui:       ╵        ~~~~~~~~~~~~~~~
@penrose/browser-ui:  > src/inspector/makeViewBoxes.tsx:1:45: error: Could not resolve "@penrose/core" (mark it as external to exclude it from the bundle)
@penrose/browser-ui:     1 │ import { bBoxDims, RenderShape, Shape } from "@penrose/core";
@penrose/browser-ui:       ╵                                              ~~~~~~~~~~~~~~~
@penrose/browser-ui: Unhandled exception: TypeError: The feature watch recursively is unavailable on the current platform, which is being used to run Node.js
@penrose/browser-ui:     at new NodeError (node:internal/errors:371:5)
@penrose/browser-ui:     at Object.fs [as watch] (node:fs:2247:11)
@penrose/browser-ui:     at zb.watch (/home/alexander/penrose/node_modules/serve-http/src/livereload.js:158:20)
@penrose/browser-ui:     at nb.<anonymous> (/home/alexander/penrose/node_modules/serve-http/src/server.js:139:7)
@penrose/browser-ui:     at Object.onceWrapper (node:events:639:28)
@penrose/browser-ui:     at nb.emit (node:events:520:28)
@penrose/browser-ui:     at Server.emit (node:events:520:28)
@penrose/browser-ui:     at emitListeningNT (node:net:1364:10)
@penrose/browser-ui:     at processTicksAndRejections (node:internal/process/task_queues:82:21)
@penrose/core: $ nearleyc src/parser/Domain.ne > src/parser/DomainParser.ts && nearleyc src/parser/Substance.ne > src/parser/SubstanceParser.ts && nearleyc src/parser/Style.ne > src/parser/StyleParser.ts
@penrose/browser-ui: error Command failed with exit code 2.
@penrose/browser-ui: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run watch exited 2 in '@penrose/browser-ui'
lerna WARN complete Waiting for 1 child process to exit. CTRL-C to exit immediately.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```
The tutorial says to run it again, however that does not work. A lower version does not recognize "rmSync" which is supposedly a feature added in node.js 14.
**To Reproduce**

Steps to reproduce the behavior:

1. Clone the repository, main branch, with git clone, on Linux (currently using Debian).
2. Cd into the directory, and do ``yarn && yarn start``
3. Wait until it exits with code 2.
4. See error

**Expected behavior**
``yarn start`` correctly starts on Linux
**Desktop (please complete the following information if applicable):**

- OS: Debian 11

**Additional context**
I think this is due to a problem with node not supporting recursively watching directories on Linux. It seems like others have had this problem before:
https://stackoverflow.com/questions/43112227/recursive-node-js-fs-watch-on-linux#46513049
https://github.com/gmasmejean/watch-recursivly/blob/master/watch.js
https://github.com/nodejs/node/issues/36005

--- a/package.json
+++ b/package.json
@@ -10,6 +10,7 @@
     "clean": "lerna clean && lerna run clean --stream",
     "prestart": "lerna run build --scope=@penrose/examples --scope=@penrose/core",
     "start": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/browser-ui",
+    "prestart:ide": "lerna run build --scope=@penrose/examples --scope=@penrose/core",
     "start:ide": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/panels",
     "build:ide": "lerna run build --stream --scope=@penrose/core --scope=@penrose/panels  --include-dependencies",
     "build:docs-site": "lerna run build --stream --scope=@penrose/core --scope=@penrose/components --scope=@penrose/docs-site",

--- a/package.json
+++ b/package.json
@@ -8,6 +8,7 @@
   "scripts": {
     "build": "lerna run build --stream --concurrency 1",
     "clean": "lerna clean && lerna run clean --stream",
+    "prestart": "lerna run build --scope=@penrose/core",
     "start": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/browser-ui",
     "start:ide": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/panels",
     "build:ide": "lerna run build --stream --scope=@penrose/core --scope=@penrose/panels  --include-dependencies",

--- a/package.json
+++ b/package.json
@@ -8,7 +8,7 @@
   "scripts": {
     "build": "lerna run build --stream --concurrency 1",
     "clean": "lerna clean && lerna run clean --stream",
-    "prestart": "lerna run build --scope=@penrose/core",
+    "prestart": "lerna run build --scope=@penrose/examples --scope=@penrose/core",
     "start": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/browser-ui",
     "start:ide": "lerna run watch --stream --parallel --scope=@penrose/core --scope=@penrose/panels",
     "build:ide": "lerna run build --stream --scope=@penrose/core --scope=@penrose/panels  --include-dependencies",

--- a/packages/browser-ui/index.html
+++ b/packages/browser-ui/index.html
@@ -1,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/src/favicon.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Penrose</title>
+    <link
+      href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700"
+      rel="stylesheet"
+    />
+  </head>
+  <body>
+    <div id="root"></div>
+    <script type="module" src="/src/index.tsx"></script>
+  </body>
+</html>

--- a/packages/browser-ui/index.html
+++ b/packages/browser-ui/index.html
@@ -12,6 +12,10 @@
   </head>
   <body>
     <div id="root"></div>
+    <script type="text/javascript">
+      // HACK for eigen dep which uses `global`
+      window.global = window;
+    </script>
     <script type="module" src="/src/index.tsx"></script>
   </body>
 </html>

--- a/packages/browser-ui/package.json
+++ b/packages/browser-ui/package.json
@@ -37,8 +37,9 @@
     "/build"
   ],
   "scripts": {
-    "watch": "node build.js -w",
-    "start": "yarn watch",
+    "dev": "vite",
+    "watch": "vite",
+    "start": "vite",
     "build": "node build.js",
     "test": "echo NO TESTS",
     "lint": "eslint --ext js,ts,tsx src",
@@ -72,7 +73,9 @@
     "ttypescript": "^1.5.12",
     "typescript": "^4.1.3",
     "typescript-transform-paths": "^2.0.1",
-    "webpack-cli": "^4.2.0"
+    "webpack-cli": "^4.2.0",
+    "vite": "^2.8.0",
+    "@vitejs/plugin-react": "^1.0.7"
   },
   "browserslist": [
     ">0.2%",

--- a/packages/browser-ui/package.json
+++ b/packages/browser-ui/package.json
@@ -64,7 +64,7 @@
     "eslint-config-prettier": "^6.13.0",
     "eslint-plugin-jsdoc": "^30.7.3",
     "eslint-plugin-prefer-arrow": "^1.2.2",
-    "estrella": "^1.3.0",
+    "estrella": "^1.4.1",
     "jest-junit": "^12.0.0",
     "serve-http": "^1.0.6",
     "terser": "^5.5.1",

--- a/packages/browser-ui/src/App.tsx
+++ b/packages/browser-ui/src/App.tsx
@@ -15,12 +15,12 @@ import {
 } from "@penrose/core";
 import animalNameList from "animals";
 import colorNameList from "color-name-list";
-import Inspector from "inspector/Inspector";
 import { isEqual } from "lodash";
 import * as React from "react";
 import SplitPane from "react-split-pane";
-import ButtonBar from "ui/ButtonBar";
-import { FileSocket, FileSocketResult } from "ui/FileSocket";
+import Inspector from "./inspector/Inspector";
+import ButtonBar from "./ui/ButtonBar";
+import { FileSocket, FileSocketResult } from "./ui/FileSocket";
 
 //#region variation generation
 

--- a/packages/browser-ui/src/inspector/views/ShapeView.tsx
+++ b/packages/browser-ui/src/inspector/views/ShapeView.tsx
@@ -1,6 +1,6 @@
-import makeViewBoxes from "inspector/makeViewBoxes";
 import * as React from "react";
 import { ObjectInspector } from "react-inspector";
+import makeViewBoxes from "../../inspector/makeViewBoxes";
 import IViewProps from "./IViewProps";
 
 interface IState {

--- a/packages/browser-ui/src/inspector/views/viewMap.tsx
+++ b/packages/browser-ui/src/inspector/views/viewMap.tsx
@@ -1,5 +1,5 @@
-import Errors from "inspector/views/Errors";
 //import Opt from "./Opt";
+import Errors from "../../inspector/views/Errors";
 import CompGraph from "./CompGraph";
 import Frames from "./Frames";
 import Settings from "./Settings";

--- a/packages/browser-ui/src/vite-env.d.ts
+++ b/packages/browser-ui/src/vite-env.d.ts
@@ -1,0 +1,1 @@
+/// <reference types="vite/client" />

--- a/packages/browser-ui/vite.config.js
+++ b/packages/browser-ui/vite.config.js
@@ -4,9 +4,6 @@ import { defineConfig } from "vite";
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [react()],
-  define: {
-    global: "window",
-  },
   json: {
     stringify: true,
   },

--- a/packages/browser-ui/vite.config.js
+++ b/packages/browser-ui/vite.config.js
@@ -1,0 +1,24 @@
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  define: {
+    global: "window",
+  },
+  json: {
+    stringify: true,
+  },
+  build: {
+    sourcemap: true,
+    rollupOptions: {
+      external: ["react", "react-dom"],
+      output: {
+        globals: {
+          react: "React",
+        },
+      },
+    },
+  },
+});

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -77,7 +77,7 @@
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-jsdoc": "^30.7.3",
     "eslint-plugin-react": "^7.28.0",
-    "estrella": "^1.3.0",
+    "estrella": "^1.4.1",
     "jest": "^27.5.1",
     "jest-bench": "^27.5.1",
     "jscodeshift": "^0.11.0",

--- a/packages/panels/index.html
+++ b/packages/panels/index.html
@@ -16,7 +16,7 @@
       href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700"
       rel="stylesheet"
     />
-    <link href="./app.css" rel="stylesheet" />
+    <link href="/src/App.css" rel="stylesheet" />
 
     <!--
       Notice the use of %PUBLIC_URL% in the tags above.
@@ -33,7 +33,7 @@
   <body>
     <noscript> You need to enable JavaScript to run this app. </noscript>
     <div id="root"></div>
-    <script type="text/javascript" src="./app.js"></script>
+    <script type="module" src="/src/App.tsx"></script>
     <!--
       This HTML file is a template.
       If you open it directly in the browser, you will see an empty page.

--- a/packages/panels/index.html
+++ b/packages/panels/index.html
@@ -33,7 +33,11 @@
   <body>
     <noscript> You need to enable JavaScript to run this app. </noscript>
     <div id="root"></div>
-    <script type="module" src="/src/App.tsx"></script>
+    <script type="text/javascript">
+      // HACK for eigen dep which uses `global`
+      window.global = window;
+    </script>
+    <script type="module" src="/src/index.tsx"></script>
     <!--
       This HTML file is a template.
       If you open it directly in the browser, you will see an empty page.

--- a/packages/panels/package.json
+++ b/packages/panels/package.json
@@ -54,7 +54,7 @@
     "@types/react-panelgroup": "^1.0.1",
     "@types/react-router-dom": "^5.1.7",
     "@types/styled-components": "^5.1.7",
-    "estrella": "^1.3.0",
+    "estrella": "^1.4.1",
     "typescript": "^4.1.3",
     "web-vitals": "^0.2.4"
   }

--- a/packages/panels/package.json
+++ b/packages/panels/package.json
@@ -5,10 +5,10 @@
   "license": "MIT",
   "dependencies": {
     "@monaco-editor/react": "^4.0.11",
-    "monaco-editor": "^0.22.3",
     "@octokit/rest": "^18.3.2",
     "@penrose/core": "^1.3.0",
     "lodash": "^4.17.21",
+    "monaco-editor": "^0.22.3",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-router-dom": "^5.2.0",
@@ -17,8 +17,9 @@
     "styled-components": "4.4.1"
   },
   "scripts": {
-    "watch": "node build.js -w",
-    "start": "node build.js -w",
+    "dev": "vite",
+    "watch": "vite",
+    "start": "vite",
     "build": "node build.js",
     "test": "echo NO TESTS",
     "prepack": "yarn build",
@@ -43,6 +44,7 @@
     ]
   },
   "devDependencies": {
+    "@esbuild-plugins/node-modules-polyfill": "^0.1.4",
     "@testing-library/jest-dom": "^5.11.6",
     "@testing-library/react": "^11.2.2",
     "@testing-library/user-event": "^12.6.0",

--- a/packages/panels/package.json
+++ b/packages/panels/package.json
@@ -17,9 +17,8 @@
     "styled-components": "4.4.1"
   },
   "scripts": {
-    "dev": "vite",
-    "watch": "vite",
-    "start": "vite",
+    "watch": "node build.js -w",
+    "start": "node build.js -w",
     "build": "node build.js",
     "test": "echo NO TESTS",
     "prepack": "yarn build",

--- a/packages/panels/package.json
+++ b/packages/panels/package.json
@@ -17,8 +17,9 @@
     "styled-components": "4.4.1"
   },
   "scripts": {
-    "start": "yarn watch",
-    "watch": "node build.js -w",
+    "dev": "vite",
+    "watch": "vite",
+    "start": "vite",
     "build": "node build.js",
     "test": "echo NO TESTS",
     "prepack": "yarn build",

--- a/packages/panels/vite.config.js
+++ b/packages/panels/vite.config.js
@@ -1,14 +1,20 @@
+import { NodeModulesPolyfillPlugin } from "@esbuild-plugins/node-modules-polyfill";
 import react from "@vitejs/plugin-react";
 import { defineConfig } from "vite";
 
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [react()],
-  define: {
-    global: "window",
+  resolve: {
+    alias: {
+      stream: "rollup-plugin-node-polyfills/polyfills/stream",
+    },
   },
-  json: {
-    stringify: true,
+  optimizeDeps: {
+    esbuildOptions: {
+      // Enable esbuild polyfill plugins
+      plugins: [NodeModulesPolyfillPlugin()],
+    },
   },
   build: {
     sourcemap: true,

--- a/packages/panels/vite.config.js
+++ b/packages/panels/vite.config.js
@@ -1,0 +1,24 @@
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react()],
+  define: {
+    global: "window",
+  },
+  json: {
+    stringify: true,
+  },
+  build: {
+    sourcemap: true,
+    rollupOptions: {
+      external: ["react", "react-dom"],
+      output: {
+        globals: {
+          react: "React",
+        },
+      },
+    },
+  },
+});

